tools/dissector/sample_dissector.cpp

```diff
@@ -505,54 +938,87 @@ namespace OpenDDS
     }
 
     size_t
-    Sample_Sequence::dissect_i (Wireshark_Bundle_i &params,
-                                const std::string &label)
+    Sample_Sequence::dissect_i (Wireshark_Bundle &params)
     {
-      guint8 *data = params.get_remainder();
-      guint32 len = (guint32)compute_length (data);
-      guint32 count = *(reinterpret_cast< guint32 * >(data));
+      size_t len = 0;
+      if (!params.get_size_only) {
+        len = compute_length(params);
+      }
 
-      std::stringstream outstream;
-      outstream << label;
-      if (params.use_index)
-        outstream << "[" << params.index << "]";
-      outstream << " (length = " << count << ")";
-      proto_item *item =
-        proto_tree_add_text (params.tree, params.tvb, params.offset,
-                             len,"%s", outstream.str().c_str());
-      proto_tree *subtree = proto_item_add_subtree (item, ett_payload_);
-
-      size_t data_pos = 4;
-      Wireshark_Bundle_i sp = params;
-      sp.tree = subtree;
-      for (guint32 ndx = 0; ndx < count; ndx++)
-        {
-          sp.index = ndx;
-          sp.use_index = true;
-          sp.offset = params.offset + (gint)data_pos;
-          data_pos += element_->dissect_i (sp, label);
-        }
-      return data_pos;
-    }
+      size_t location = params.buffer_pos();
 
-    size_t
-    Sample_Sequence::compute_length (guint8 *data)
-    {
+      // Get Count of Elements
+      ACE_CDR::ULong count;
+      params.serializer >> count;
+      size_t count_size = params.buffer_pos() - location;
+      if (params.get_size_only) {
+        len = count_size;
+      }
 
-      size_t len = 4;
-      guint32 count = *(reinterpret_cast< guint32 * >(data));
-      data += 4;
-      while (count > 0)
-        {
-          size_t elen = element_->compute_length(data);
-          data += elen;
-          len += elen;
-          --count;
+      // Add Item
```

Comment:
There should be a lot of common code between Sequence and Array.  After sequence has read the length from the buffer, can this delegate to a method shared by Sequence and Array (possibly in a base class)?